### PR TITLE
fixes #3183 - Fixes the anonymous reporting through env

### LINKED ISSUE #3183: Investigate why anonymous reporting is enabled on staging
Currently on staging we have the following `STAGING_ANON_REPORTING_ENABLED=False`, yet anonymous reporting is enabled.

I've verified on localhost, changing the value of `ANON_REPORTING_ENABLED` to True/False has the expected behavior.

### PR DESCRIPTION
**Pull request by karlcow**
This PR fixes issue #3183

## Proposed PR background

it will change the environment variable so the requirements is for anonymous reporting to be working to have 

`ANONYMOUS_REPORTING = ON`

### CODE REVIEW

**Comment by karlcow** [comment]
At the build is failing, and probably because of 
https://github.com/pallets/werkzeug/issues/1714

```
Traceback (most recent call last):
  File "/home/circleci/repo/env/lib/python3.7/site-packages/nose/failure.py", line 39, in runTest
    raise self.exc_val.with_traceback(self.tb)
  File "/home/circleci/repo/env/lib/python3.7/site-packages/nose/loader.py", line 418, in loadTestsFromName
    addr.filename, addr.module)
  File "/home/circleci/repo/env/lib/python3.7/site-packages/nose/importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/home/circleci/repo/env/lib/python3.7/site-packages/nose/importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/usr/local/lib/python3.7/imp.py", line 244, in load_module
    return load_package(name, filename)
  File "/usr/local/lib/python3.7/imp.py", line 216, in load_package
    return _load(spec)
  File "<frozen importlib._bootstrap>", line 696, in _load
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/circleci/repo/config/__init__.py", line 254, in <module>
    from webcompat import app  # noqa
  File "/home/circleci/repo/webcompat/__init__.py", line 35, in <module>
    import webcompat.views  # noqa
  File "/home/circleci/repo/webcompat/views.py", line 23, in <module>
    from webcompat.form import AUTH_REPORT
```

**Comment by karlcow** [comment]
ok we need to bump some versions
https://requires.io/github/webcompat/webcompat.com/requirements/?branch=master

Here it is
https://github.com/lepture/flask-wtf/compare/v0.14.2...0.14.3#diff-ba558d997f811e655c9dd6f6e2f91f3f

**Comment by karlcow** [comment]
https://github.com/pallets/werkzeug/issues/1714

**Comment by karlcow** [comment]
I fixed the dependency on werkzeug, so CircleCI can run. 
And we will fix this in #3187 the dependencies on more modern Werkzeug.

**Comment by karlcow** [comment]
oh sweeeet…
https://app.circleci.com/jobs/github/webcompat/webcompat.com/1010


```
Done.
...............................................................................................................F.....................................
======================================================================
FAIL: Test that anonymous post succeeds or not on /issues/new.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/circleci/repo/env/lib/python3.7/site-packages/mock/mock.py", line 1330, in patched
    return func(*args, **keywargs)
  File "/home/circleci/repo/tests/unit/test_urls.py", line 95, in test_successful_post_new_issue
    self.assertEqual(rv.status_code, 400)
AssertionError: 302 != 400
-------------------- >> begin captured logging << --------------------
webcompat: INFO: None b'http://testing.example.org'
webcompat: INFO: None b'http://testing.example.org'
--------------------- >> end captured logging << ---------------------

----------------------------------------------------------------------
Ran 149 tests in 1.290s

FAILED (failures=1)
```

**Comment by karlcow** [comment]
at least it is now related to anonymous reporting test. I'll look at this on monday except if someone beats me to it.

**Comment by karlcow** [comment]
ok understood. 
I changed a bit the test to see what it was happening with a real request.

```
    def test_fail_anonymous_post_new_issue(self):
        """Test that anonymous post fail when this is off."""
        webcompat.app.config['ANONYMOUS_REPORTING'] = 'OFF'
        rv = self.app.post(
            '/issues/new',
            content_type='multipart/form-data',
            environ_base=headers,
            data=dict(
                browser='Firefox Mobile 45.0',
                description='testing 2971',
                os='macos',
                problem_category='yada',
                submit_type='github-proxy-report',
                url='http://testing.example.org',
                username='yeeha'))
        self.assertEqual(rv.status_code, 400)
```

It created https://github.com/webcompat/webcompat-tests-private/issues/55
and https://github.com/webcompat/webcompat-tests/issues/2378

because `ANONYMOUS_REPORTING` is blocking the UI but not really the API.

we need to fence the api too.

**Comment by karlcow** [comment]
ok I think I found.

File: config/environment.py [hunk 1]
```diff
@@ -65,7 +65,7 @@
     OAUTH_TOKEN = os.environ.get('OAUTH_TOKEN')
     SECRET_KEY = os.environ.get('SECRET_KEY') or 'NO SECRETS'
     HOOK_SECRET_KEY = os.environ.get('HOOK_SECRET_KEY') or 'SECRETS'
-    ANON_REPORTING_ENABLED = os.environ.get('ANON_REPORTING_ENABLED') or False
+    ANONYMOUS_REPORTING = os.environ.get('ANONYMOUS_REPORTING') or False
```
Comment (ksy36) [inline config/environment.py]: I'm late since it's been merged already, but this should probably be `or 'OFF'`

Comment (miketaylr) [reply config/environment.py]: Oh... you're right. :)

I'll send in a PR, because it was my bad for merging too early. 🙈

**Comment by miketaylr** [comment]
I've verified on staging that this does what we expect it to.

**Pull request merged**